src/main/java/pl/futurecollars/invoicing/db/file/FileBasedDatabase.java

```diff
@@ -6,11 +6,15 @@
 import java.util.Optional;
 import java.util.stream.Collectors;
 import lombok.AllArgsConstructor;
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Repository;
 import pl.futurecollars.invoicing.db.Database;
 import pl.futurecollars.invoicing.model.Invoice;
 import pl.futurecollars.invoicing.util.FilesService;
 import pl.futurecollars.invoicing.util.JsonService;
 
+@Repository
+@Primary
```

Comment:
Jeśli tworzysz beana w klasie DatabaseConfig (adnotacją @Bean) to nie potrzebujesz już @Repository, ta klasa może być zwykłą klasą bez adnotacji. Primary też wystarczy na samej metodzie z @Bean - tu masz w obu miejsach